test/evp_fetch_prov_test.c

```diff
@@ -73,16 +76,38 @@ static int calculate_digest(const EVP_MD *md, const char *msg, size_t len,
     return ret;
 }
 
+static int test_fip_mode(OPENSSL_CTX *ctx, int loaded_fips)
+{
+    return TEST_int_eq(FIPS_mode_with_ctx(ctx), 0)
+           && TEST_int_eq(FIPS_mode_set_with_ctx(ctx, 1), loaded_fips)
+           && TEST_int_eq(FIPS_mode_with_ctx(ctx), 1)
```

Comment:
If `FIPS_mode_set_with_ctx` fails, then I would expect `FIPS_mode_with_ctx` to return 0. But here it seems to return 1 regardless.

Reply:
Ok changed the code to restore the old state in this case.